sematic/db/models/job.py

```diff
@@ -0,0 +1,151 @@
+# Standard Library
+import datetime
+from dataclasses import asdict
+from typing import Any, Dict, List, Sequence, Tuple, Union
+
+# Third-party
+from sqlalchemy import Column, types
+
+# Sematic
+from sematic.db.models.base import Base
+from sematic.db.models.mixins.json_encodable_mixin import JSONEncodableMixin
+from sematic.scheduling.job_details import (
+    JobDetails,
+    JobKindString,
+    JobStatus,
+    KubernetesJobStateString,
+)
+from sematic.types.types.dataclass import fromdict
+from sematic.utils.exceptions import IllegalStateTransitionError
+
+
+class Job(Base, JSONEncodableMixin):
+    """Represents a K8s job managed by Sematic.
+
+    This holds high-level metadata about the job, as well as its status history.
+    The metadata here is primarily useful to the rest of Sematic. It contains a
+    detail_serialization which can contain lots of information about the Kubernetes
+    job & related pod(s).
+
+    Attributes
+    ----------
+    name:
+        The k8s name of the job. Used instead of an 'id' field because
+        K8s identifies jobs by namespace and name.
+    namespace:
+        The k8s namespace of the job. In conjunction with name, forms
+        the unique id for the job.
+    run_id:
+        The id of the run the job is associated with.
+    last_updated_epoch_seconds:
+        The time the job status was last synced with Kubernetes. Pulled from
+        the most recent status to aid in queries.
+    state:
+        The simple state of the job. Pulled from the most recent status to
+        aid in queries.
+    kind:
+        Whether the job is for executing a run or a resolution.
+    message:
+        The human-readable message for the latest status. Pulled from
+        the most recent status to aid in queries.
+    detail_serialization:
+        A json-encoded serialization of detailed state information for the job
+    status_history_serialization:
+        A list of json-encoded serializations of job statuses
```

Comment:
Please specify that the statuses are in chronological order.

Reply:
done